### DISCUSSION TITLE
Prisma tutorial - field password missing?

### DISCUSSION THREAD

**Discussion by TomasMoskala** [prisma/prisma#2737] [Q&A] [answered]
Hi, 
is the field password missing here or am I missing something?
Page
https://www.prisma.io/tutorials/authentication-in-apollo-server-ct21

Step 2
Open src/typeDefs.js and paste the following schema between the gql tags:

```
type User {
  id: ID!
  username: String!
}

type Query {
  currentUser: User!
}

type Mutation {
  register(username: String!, password: String!): User!
  login(username: String!, password: String!): LoginResponse!
}

type LoginResponse {
  token: String
  user: User
}
```
Here it comes

```
const resolvers = {
  Mutation: {
    register: async (parent, { username, password }, ctx, info) => {
      const hashedPassword = await bcrypt.hash(password, 10)
      const user = await ctx.prisma.createUser({
        username,
        password: hashedPassword,
      })
      return user
    },
  },
}
```
And here
```
login: async (parent, { username, password }, ctx, info) => {
  const user = await ctx.prisma.user({ username })

  if (!user) {
    throw new Error('Invalid Login')
  }

  const passwordMatch = await bcrypt.compare(password, user.password)

  if (!passwordMatch) {
    throw new Error('Invalid Login')
  }

  const token = jwt.sign(
    {
      id: user.id,
      username: user.email,
    },
    'my-secret-from-env-file-in-prod',
    {
      expiresIn: '30d', // token will expire in 30days
    },
  )
  return {
    token,
    user,
  }
}
```

**Comment by williamluke4** [comment, marked as answer]
This is a very old repo, which is probably using Prisma 1. The password field is not missing, it is [stored](https://github.com/pantharshit00/apollo-server-prisma-starter/blob/master/datamodel.prisma) in thedatabase but not exposed through GraphQL (For Obvious Reasons) 

 Have a look at https://redwoodjs.com/

**Reply by TomasMoskala** [reply]
Thanks, is there any new repo where I could find an example of Prisma+Nexus+Apollo server+jwt login+this way protected resolver?
I am stuggling quite a lot with this as all the examples and tutorials are outdated. I am not sure how to put the user into context and what is the relation between schema context in makeSchema function and request context in the server instance.

**Reply by ryands17** [reply]
@TomasMoskala have a look at these [examples](https://github.com/prisma/prisma-examples/) for Prisma. There are ones with Apollo Server and authentication as well.

**Comment by ludralph** [comment]
Hi @TomasMoskala,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
